modules/compute/gke-node-pool/outputs.tf

```diff
@@ -14,25 +14,26 @@
   * limitations under the License.
   */
 
-output "node_pool_name" {
+output "node_pool_names" {
   description = "Name of the node pool."
-  value       = google_container_node_pool.node_pool.name
+  value       = google_container_node_pool.node_pool[*].name
 }
 
 locals {
   is_a_series = local.machine_family == "a2"
   last_digit  = trimsuffix(try(local.machine_vals[2], 0), "g")
 
   # Shared core machines only have 1 cpu allocatable, even if they have 2 cpu capacity
-  vcpu        = local.machine_shared_core ? 1 : local.is_a_series ? local.last_digit * 12 : local.last_digit
-  useable_cpu = local.set_threads_per_core ? local.threads_per_core * local.vcpu / 2 : local.vcpu
+  # Logic updated to work with TPUs
+  vcpu        = local.has_gpu ? (local.machine_shared_core ? 1 : local.is_a_series ? local.last_digit * 12 : local.last_digit) : 0
+  useable_cpu = local.has_gpu ? (local.set_threads_per_core ? local.threads_per_core * local.vcpu / 2 : local.vcpu) : 0
```

Comment:
This change is not needed. local.vcpu would be 0 for tpu. We could keep the old logic as it is.